### ISSUE TITLE
needless_doctest_main false positive with ```txt

### ISSUE THREAD

**Issue by Luro02** [rust-lang/rust-clippy#5280]
This piece of documentation triggers the lint
```rust
/// ```txt
/// 1 |   fn main() {
///   |          __ ^
///   |  _________|_|
///   | |         |
///   | |         this is a pair of parenthesis
/// 2 | |     println!("Hello World!")
///   | |             |^^^^^^^^^^^^^^| this is a pair of parenthesis
///   | |             |_____________||
///   | |                           |
///   | |                           this is a string. Hard to see where it starts, uh?
/// 3 | | }
///   | |_^ this is a pair of braces
/// ```
```

```
warning: needless `fn main` in doctest
   --> src/fmt.rs:207:4
    |
207 | /// 1 |   fn main() {
    |    ^^^^^^^^^^^^^^^^^^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_doctest_main
```

[`playground`](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&code=%23!%5Bdeny(clippy%3A%3Aneedless_doctest_main)%5D%0A%0A%2F%2F%2F%20%60%60%60txt%0A%2F%2F%2F%201%20%7C%20%20%20fn%20main()%20%7B%0A%2F%2F%2F%20%20%20%7C%20%20%20%20%20%20%20%20%20%20__%20%5E%0A%2F%2F%2F%20%20%20%7C%20%20_________%7C_%7C%0A%2F%2F%2F%20%20%20%7C%20%7C%20%20%20%20%20%20%20%20%20%7C%0A%2F%2F%2F%20%20%20%7C%20%7C%20%20%20%20%20%20%20%20%20this%20is%20a%20pair%20of%20parenthesis%0A%2F%2F%2F%202%20%7C%20%7C%20%20%20%20%20println!(%22Hello%20World!%22)%0A%2F%2F%2F%20%20%20%7C%20%7C%20%20%20%20%20%20%20%20%20%20%20%20%20%7C%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%7C%20this%20is%20a%20pair%20of%20parenthesis%0A%2F%2F%2F%20%20%20%7C%20%7C%20%20%20%20%20%20%20%20%20%20%20%20%20%7C_____________%7C%7C%0A%2F%2F%2F%20%20%20%7C%20%7C%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%7C%0A%2F%2F%2F%20%20%20%7C%20%7C%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20this%20is%20a%20string.%20Hard%20to%20see%20where%20it%20starts%2C%20uh%3F%0A%2F%2F%2F%203%20%7C%20%7C%20%7D%0A%2F%2F%2F%20%20%20%7C%20%7C_%5E%20this%20is%20a%20pair%20of%20braces%0A%2F%2F%2F%20%60%60%60%0Apub%20fn%20example()%20%7B%7D%0A%0Afn%20main()%20%7B%7D%0A)

**Comment by flip1995** [comment]
@llogiq can we trigger this lint only on rust code blocks?

**Comment by llogiq** [comment]
Yeah, I can do that.
